src/commonMain/kotlin/org/cru/godtools/tool/model/Multiselect.kt

```diff
@@ -0,0 +1,99 @@
+package org.cru.godtools.tool.model
+
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.map
+import org.cru.godtools.tool.FEATURE_MULTISELECT
+import org.cru.godtools.tool.ParserConfig
+import org.cru.godtools.tool.internal.RestrictTo
+import org.cru.godtools.tool.internal.VisibleForTesting
+import org.cru.godtools.tool.state.State
+import org.cru.godtools.tool.xml.XmlPullParser
+import org.cru.godtools.tool.xml.parseChildren
+
+private const val XML_STATE = "state"
+private const val XML_SELECTION_LIMIT = "selection-limit"
+private const val XML_OPTION = "option"
+private const val XML_OPTION_VALUE = "value"
+
+class Multiselect : Content {
+    internal companion object {
+        internal const val XML_MULTISELECT = "multiselect"
+    }
+
+    @VisibleForTesting
+    internal val stateName: String
+    @VisibleForTesting
+    internal val selectionLimit: Int
+
+    val options: List<Option>
+
+    internal constructor(parent: Base, parser: XmlPullParser) : super(parent, parser) {
+        parser.require(XmlPullParser.START_TAG, XMLNS_CONTENT, XML_MULTISELECT)
+
+        stateName = parser.getAttributeValue(XML_STATE).orEmpty()
+        selectionLimit = (parser.getAttributeValue(XML_SELECTION_LIMIT)?.toIntOrNull() ?: 1).coerceAtLeast(1)
+
+        options = mutableListOf()
+        parser.parseChildren {
+            when (parser.namespace) {
+                XMLNS_CONTENT -> when (parser.name) {
+                    XML_OPTION -> options += Option(this, parser)
+                }
+            }
+        }
+    }
+
+    @RestrictTo(RestrictTo.Scope.TESTS)
+    internal constructor(
+        parent: Base = Manifest(),
+        stateName: String = "",
+        selectionLimit: Int = 1,
+        options: ((Multiselect) -> List<Option>)? = null
+    ) : super(parent) {
+        this.stateName = stateName
+        this.selectionLimit = selectionLimit
+        this.options = options?.invoke(this).orEmpty()
+    }
+
+    override val isIgnored get() = FEATURE_MULTISELECT !in ParserConfig.supportedFeatures || super.isIgnored
```

Comment:
I just want to make sure I got this right, but this is only for the tests right? But ```isIgnored``` isn't specific for the test.

Reply:
Also what's the point of ```isIgnored```? Does the multiselect not get rendered if ```isIgnored``` is true?

Reply:
yes, the `Multiselect` element is ignored by the parser/renderer when this method returns true.

The parent implementation checks the `restrictTo` attribute to determine if this element is for a specific platform or not. The `supportedFeatures` check was added because we need to be able to use the same parser for Android and iOS which both will implement the view logic independently.

So, as I'm working on the Android implementation I enable `FEATURE_MULTISEELCT` and will have the element in my model hierarchy. meanwhile iOS can leave `FEATURE_MULTISELECT` disabled and not have to worry about the element yet.